src/core_plugins/kibana/public/management/sections/objects/components/objects_table/objects_table.js

```diff
@@ -0,0 +1,314 @@
+import { scanAllTypes } from '../../lib/scan_all_types';
+import { saveToFile } from '../../lib/save_to_file';
+import { getQueryText } from '../../lib/get_query_text';
+import { getSavedObjectIcon } from '../../lib/get_saved_object_icon';
+import { ensureMinimumTime } from '../../../indices/create_index_pattern_wizard/lib/ensure_minimum_time';
+import { getRelationships } from '../../lib/get_relationships';
+
+export const INCLUDED_TYPES = ['index-pattern', 'visualization', 'dashboard', 'search'];
+
+export class ObjectsTable extends Component {
+  static propTypes = {
+    savedObjectsClient: PropTypes.object.isRequired,
+    indexPatterns: PropTypes.object.isRequired,
+    $http: PropTypes.func.isRequired,
+    basePath: PropTypes.string.isRequired,
+    newIndexPatternUrl: PropTypes.string.isRequired,
+    kbnIndex: PropTypes.string.isRequired,
+    services: PropTypes.array.isRequired,
+    getEditUrl: PropTypes.func.isRequired,
+    goInApp: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      totalCount: 0,
+      page: 0,
+      perPage: 10,
+      savedObjects: [],
+      savedObjectCounts: INCLUDED_TYPES.reduce((accum, type) => {
+        accum[type] = 0;
+        return accum;
+      }, {}),
+      activeQuery: Query.parse(''),
+      selectedSavedObjects: [],
+      isShowingImportFlyout: false,
+      isSearching: false,
+      totalItemCount: 0,
+      isShowingRelationships: false,
+      relationshipId: undefined,
+      relationshipType: undefined,
+      relationshipTitle: undefined,
+    };
+  }
+
+  componentWillMount() {
+    this.fetchSavedObjects();
+    this.fetchCounts();
+  }
+
+  fetchCounts = async () => {
+    const fetches = INCLUDED_TYPES.map(type => this.props.savedObjectsClient.find({
+      perPage: 1,
+      type,
+      page: 1,
+      fields: ['id'],
+    }));
+    const result = await Promise.all(fetches);
+
+    const savedObjectCounts = result.reduce((accum, { total, savedObjects }) => {
+      if (savedObjects && savedObjects.length) {
+        accum[savedObjects[0].type] = total;
+      }
+      return accum;
+    }, this.state.savedObjectCounts);
+
+    this.setState({ savedObjectCounts });
+  }
+
+  fetchSavedObjects = async () => {
+    const { savedObjectsClient } = this.props;
+    const { activeQuery, page, perPage } = this.state;
+
+    if (!activeQuery) {
+      return {
+        pageOfItems: [],
+        totalItemCount: 0,
+      };
+    }
+
+    this.setState({ isSearching: true });
+
+    const queryText = getQueryText(activeQuery);
+    const visibleTypes =
+      activeQuery && activeQuery.ast.getFieldClauses('type')
+        ? activeQuery.ast.getFieldClauses('type')[0].value
+        : undefined;
+
+    let savedObjects = [];
+    let totalItemCount = 0;
+
+    const includeTypes = INCLUDED_TYPES.filter(type => !visibleTypes || visibleTypes.includes(type));
+
+    // TODO: is there a good way to stop existing calls if the input changes?
```

Comment:
Not that I know of with the existing APIs. But it may be worth hooking this up to some sort of Promise or timer so that you can tail them instead of arbitrarily firing them (and thus kill anything that hasn't already fired).